type_inference_visitor.py

```diff
@@ -0,0 +1,93 @@
+from astroid.transforms import TransformVisitor
+import astroid
+import warnings
+
+
+class TypeVisitor(TransformVisitor):
+
+    def _visit(self, node):
+        # If node is an instance of List/Tuple/Dict, instead of breaking
+        # down to smaller pieces of node such as const nodes, keep the
+        # bigger piece and directly use an transform function on it.
+        if isinstance(node, astroid.List) or isinstance(node, astroid.Tuple)\
+                or isinstance(node, astroid.Dict):
+            return self._transform(node)
+        if hasattr(node, '_astroid_fields'):
+            for field in node._astroid_fields:
+                value = getattr(node, field)
+                visited = self._visit_generic(value)
+                setattr(node, field, visited)
+        return self._transform(node)
+
+    def _transform(self, node):
+        """Call matching transforms for the given node if any and return the
+        transformed node.
+        """
+        node_class = node.__class__
+        if node_class not in self.transforms:
```

Comment:
I believe we already discussed this, but I don't think this block is necessary. The whole point of the `register_transform` is to tell the visitor what nodes to perform actions on.